### ISSUE TITLE
Cannot read properties of undefined (reading 'refrefsh) on guild.members.fetch()

### ISSUE THREAD

**Issue by lilabyte** [discordjs/discord.js#6980]
### Issue description

When I'm trying to use `guild.members.fetch()` this error is thrown at first:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'refresh')
/home/node_modules/discord.js/src/managers/GuildMemberManager.js(417,17): handler
node:events(394,28): emit
/home/node_modules/discord.js/src/client/websocket/handlers/GUILD_MEMBERS_CHUNK.js(31,10): module.exports
/home/node_modules/discord.js/src/client/websocket/WebSocketManager.js(350,31): handlePacket
/home/node_modules/discord.js/src/client/websocket/WebSocketShard.js(443,22): onPacket
/home/node_modules/discord.js/src/client/websocket/WebSocketShard.js(300,10): onMessage
/home/node_modules/discord.js/node_modules/ws/lib/event-target.js(199,18): onMessage
node:events(394,28): emit
/home/node_modules/discord.js/node_modules/ws/lib/websocket.js(1022,20): receiverOnMessage
node:events(394,28): emit

TypeError: Cannot read properties of undefined (reading 'refresh')
```

and after some time I also got:

```
Error: Error [GUILD_MEMBERS_TIMEOUT]: Members didn't arrive in time.
    at /home/node_modules/discord.js/src/managers/GuildMemberManager.js:435:16
```

### Code sample

```typescript
const { Client, Intents } = require('discord.js')
const client = new Client({ intents: ['GUILDS', 'GUILD_MEMBERS', 'GUILD_BANS', 'GUILD_EMOJIS_AND_STICKERS', 'GUILD_INTEGRATIONS', 'GUILD_WEBHOOKS', 'GUILD_INVITES', 'GUILD_VOICE_STATES', 'GUILD_PRESENCES', 'GUILD_MESSAGES', 'GUILD_MESSAGE_REACTIONS', 'GUILD_MESSAGE_TYPING', 'DIRECT_MESSAGES', 'DIRECT_MESSAGE_REACTIONS', 'DIRECT_MESSAGE_TYPING'], fetchAllMembers: true })

client.on('ready', function() {
    client.guilds.fetch('guild_id').then(guild => {
        client.roles.fetch().catch(console.error) // Works fine
        client.members.fetch().catch(console.error) // Throws an error
    })
})

client.login('client_token')
```

### discord.js version

discord.js@13.4.0-dev.1636675518.5e0a7d5

### Node.js version

16.13.0

### Operating system

Debian 10

### Priority this issue should have

High (immediate attention needed)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GUILDS, GUILD_MEMBERS, GUILD_BANS, GUILD_EMOJIS_AND_STICKERS, GUILD_INTEGRATIONS, GUILD_WEBHOOKS, GUILD_INVITES, GUILD_VOICE_STATES, GUILD_PRESENCES, GUILD_MESSAGES, GUILD_MESSAGE_REACTIONS, GUILD_MESSAGE_TYPING, DIRECT_MESSAGES, DIRECT_MESSAGE_REACTIONS, DIRECT_MESSAGE_TYPING

### I have tested this issue on a development release

5e0a7d5

**Comment by Jiralite** [comment]
Looking at the code, this doesn't even seem possible:

https://github.com/discordjs/discord.js/blob/5e0a7d51fce3b67ba5a0e573fdc00a83693ad008/src/managers/GuildMemberManager.js#L416-L438

`timeout` is defined before `handler()` is even called unless I'm greatly missing something here.

Aside from that, your code sample won't actually work because there is no `Client#roles` property or `Client#members` property. I only assume this is a typo for the fetched guild. As a note, `fetchAllMembers` doesn't exist in the client options, so there's no use in that being there.

The timeout error you received is from not fetching the members after a long time which can imply a missing intent or not a verified application. Regardless, I'm not able to reproduce this...

**Comment by lilabyte** [comment]
> Looking at the code, this doesn't even seem possible:
> 
> https://github.com/discordjs/discord.js/blob/5e0a7d51fce3b67ba5a0e573fdc00a83693ad008/src/managers/GuildMemberManager.js#L416-L438
> 
> `timeout` is defined before `handler()` is even called unless I'm greatly missing something here.
> 
> Aside from that, your code sample won't actually work because there is no `Client#roles` property or `Client#members` property. I only assume this is a typo for the fetched guild. As a note, `fetchAllMembers` doesn't exist in the client options, so there's no use in that being there.
> 
> The timeout error you received is from not fetching the members after a long time which can imply a missing intent or not a verified application. Regardless, I'm not able to reproduce this...

Yes that was a typo. This code was started on FiveM server and it's not a problem with discord.js itself.
